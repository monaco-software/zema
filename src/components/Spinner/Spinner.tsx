import React, { FC } from 'react';

export interface SpinnerProps {
  color?: string;
  size?: number;
}

export const Spinner: FC<SpinnerProps> = ({
  color = '#7D4CDB',
  size = 28,
}) => {
  return (
    <svg
      version="1.1"
      viewBox="0 0 32 32"
      width={size}
      height={size}
      fill={color}
    >
      <path
        opacity=".25"
        d="M16 0 A16 16 0 0 0 16 32 A16 16 0 0 0 16 0 M16 4 A12 12 0 0 1 16 28 A12 12 0 0 1 16 4"
      />
      <path d="M16 0 A16 16 0 0 1 32 16 L28 16 A12 12 0 0 0 16 4z">
        <animateTransform
          attributeName="transform"
          type="rotate"
          from="0 16 16"
          to="360 16 16"
          dur="0.8s"
          repeatCount="indefinite"
        />
      </path>
    </svg>
  );
};
