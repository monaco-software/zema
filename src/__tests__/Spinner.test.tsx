import React from 'react';
import renderer from 'react-test-renderer';
import { Spinner } from '@components/Spinner/Spinner';

it('<Spinner /> should render', () => {
  const tree = renderer.create(<Spinner size={128} color="#ffffff" />).toJSON();

  expect(tree).toMatchSnapshot();
});
