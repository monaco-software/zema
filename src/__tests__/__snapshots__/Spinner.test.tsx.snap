// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> should render 1`] = `
<svg
  fill="#ffffff"
  height={128}
  version="1.1"
  viewBox="0 0 32 32"
  width={128}
>
  <path
    d="M16 0 A16 16 0 0 0 16 32 A16 16 0 0 0 16 0 M16 4 A12 12 0 0 1 16 28 A12 12 0 0 1 16 4"
    opacity=".25"
  />
  <path
    d="M16 0 A16 16 0 0 1 32 16 L28 16 A12 12 0 0 0 16 4z"
  >
    <animateTransform
      attributeName="transform"
      dur="0.8s"
      from="0 16 16"
      repeatCount="indefinite"
      to="360 16 16"
      type="rotate"
    />
  </path>
</svg>
`;
